codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ConstrainedBlobGenerator.kt

```diff
@@ -143,9 +160,22 @@ class ConstrainedBlobGenerator(
             )
         }
     }
+
+    private fun generateDisplayMessageForEachVariant() =
+        writable {
+            blobConstraintsInfo.forEach {
+                it.shapeConstraintViolationDisplayMessage(shape).invoke(this)
+            }
+        }
 }
 
-data class BlobLength(val lengthTrait: LengthTrait) {
+// Each type of constraint that can be put on a Blob must implement the BlobConstraintGenerator
+// interface. This allows the
+interface BlobConstraintGenerator {
+    fun shapeConstraintViolationDisplayMessage(shape: Shape): Writable
+}
+
```

Comment:
Do we need this interface? I see this akin to `CollectionTraitInfo` in the case of collections, which is a `sealed class`.